source/includes/code-examples/GridFS.cs

```diff
@@ -0,0 +1,415 @@
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+
+        // Downloads a file from the GridFS bucket by referencing its ObjectId value
+        // start-open-download-stream-with-options-async
+        var filter = Builders<BsonDocument>.Filter.Eq("filename", "new_file");
+        var doc = await database.GetCollection<BsonDocument>("fs.files").Find(filter).FirstOrDefaultAsync();
+        var id = doc["_id"].AsObjectId;
+
+        var options = new GridFSDownloadOptions
+        {
+            Seekable = true
+        };
+
+        using (var downloader = await bucket.OpenDownloadStreamAsync(id, options))
+        {
+            var buffer = new byte[downloader.Length];
+            await downloader.ReadAsync(buffer, 0, buffer.Length);
+
+            // Process the buffer as needed
+        }
+        // end-open-download-stream-with-options-async
+    }
+
+    static void DownloadFileWithOptions()
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+
+        // Downloads a file from the GridFS bucket by referencing its ObjectId value
+        // start-open-download-stream-with-options
+        var filter = Builders<BsonDocument>.Filter.Eq("filename", "new_file");
+        var doc = database.GetCollection<BsonDocument>("fs.files").Find(filter).FirstOrDefault();
+        var id = doc["_id"].AsObjectId;
+
+        var options = new GridFSDownloadOptions
+        {
+            Seekable = true
+        };
+
+        using (var downloader = bucket.OpenDownloadStream(id, options))
+        {
+            var buffer = new byte[downloader.Length];
+            downloader.Read(buffer, 0, buffer.Length);
+
+            // Process the buffer as needed
+        }
+        // end-open-download-stream-with-options
+    }
+    static async Task DownloadStreamAsync()
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+
+        // Downloads an entire GridFS file to a download stream
+        // start-download-to-stream-async
+        using (var outputFile = new FileStream("/path/to/output_file", FileMode.Create, FileAccess.Write))
+        {
+            var filter = Builders<BsonDocument>.Filter.Eq("filename", "new_file");
+            var doc = await database.GetCollection<BsonDocument>("fs.files").Find(filter).FirstOrDefaultAsync();
+            var id = doc["_id"].AsObjectId;
```

Comment:
I think the preceding lines can be moved outside the `using` as in previous examples.

Reply:
Waiting for the change to review it.